Introduce external .cht and CGATS file handling for colorchecker based calibration

Jiyone opened the pull request:
![image](https://github.com/user-attachments/assets/64420d78-a12c-4ed8-aeed-535f54922f24)


### SUMMARY :

The user must place matching `.cht` and CGATS files in `[config]/color/it8`.

The Color Calibration module initializes the checker list with built-in checkers, followed by any `.cht` (chart geometry) files that Ansel finds in `[config]/color/it8`.
It also builds a backend list of color definitions by loading any CGATS.17 files found in the same folder.

When the user selects a `.cht` checker from the list, Ansel compares its number of patches to those in each CGATS file from the backend list. It then populates the `Chart color` combobox with the CGATS files that have a matching patch count.
This combobox is only displayed if at least one matching file is found and a `.cht` checker is selected.

Once both a checker and a corresponding color file are selected, Ansel uses the lcms2 library to read the color data and populate the color checker structure based on the `.cht` file.

Finally, Ansel displays the chart on screen as usual.

If no CGATS files are found, a message is shown to inform the user.
![image](https://github.com/user-attachments/assets/f3486ea5-24d0-45d4-aa4b-6afd1712b1c1)



### NOTE : 
ANSI CGATS.17 is THE standard text file format for exchanging color measurement data.
This standard text format (the ASCII version is by far the most common) is the format accepted by most color measurement and profiling applications.
They can be used with lcms2.

A `.cht` file defines the geometry of a color calibration chart, such as the layout and position of its color patches. It describes how the patches are arranged (e.g., rows, columns, spacing) so that image processing software can correctly identify and extract color information from a scanned or photographed chart. It does **not** contain any color data—only the structure of the chart.

Jiyone committed:
Added IT8 ColorChecker support:

Added definitions and functions to validate and create a ColorChecker
from an IT8 file.

Jiyone committed:
refactor colorchecker code in Color Calibration, connect new code to GUI.

Jiyone committed:
wip it8

Jiyone committed:
Refactor and fix test on CGATS file compatibiliy

Jiyone committed:
Add function to find white, black and grey patch + some modifs

aurelienpierre commented on src/common/colorchecker.h:
```diff
@@ -25,343 +28,371 @@ typedef enum dt_color_checker_targets
+  .white = 18,
+  .black = 23,
+  .values
+  = { { "A1", { 37.54, 14.37, 14.92 }, { 0.087, 0.125 } },   { "A2", { 64.66, 19.27, 17.50 }, { 0.250, 0.125 } },
```
Please keep one color declaration per line, it's hard enough to read when they are stacked.

Jiyone commented on src/common/colorchecker.h:
```diff
@@ -393,15 +429,17 @@ dt_color_checker_t * dt_get_color_checker(const dt_color_checker_targets target_
 
 // get a a patch coordinates of in the checker array from the patch index in the list of values
-static inline void dt_color_checker_get_coordinates(const dt_color_checker_t *const target_checker, size_t *coordinates, const size_t index)
+static inline void dt_color_checker_get_coordinates(const dt_color_checker_t *const target_checker,
```
My VS Code does no such thing 🤔

Jiyone committed:
More init and cleanup/free function, refactor and fix

aurelienpierre commented on src/common/colorchecker.h:
```diff
@@ -381,7 +546,12 @@ dt_color_checker_t * dt_get_color_checker(const dt_color_checker_targets target_
+    case COLOR_CHECKER_USER_REF:
+      if(label_data)
+        return dt_colorchecker_user_ref_create(label_data->path);
+      
```
`break;`

Jiyone committed:
Refactor and improve IT8 color checker handling

Add material type handling and improve CGATS support.

Jiyone committed:
Added .cht files support, update the GUI for them.

Jiyone committed:
CHT : Add functions to use them

Jiyone committed:
Checker : add a ERROR message on screen if something whent wrong

Jiyone committed:
CGATS file : skip on .cht

Jiyone committed:
Deals with X and Y order when iterate patches.

And some minor code cleanup.

Jiyone committed:
fixed a typo

Jiyone committed:
Refactor colorchecker handling and improve code

- Modified functions to utilize the patch number in both .cht and checker, so the gui only shows compatible .cht files.
- Updated the chart_spec structure to initialize all fields in _dt_color_checker_chart_spec_init().
- introduced SAFE_FREE and SAFE_G_FREE macros for safer memory deallocation.
- Enhanced the _dt_colorchecker_get_standard_type function to handle invalid CGATS types more gracefully.
- Improved the update_colorchecker_cht_list function to filter .cht files based on the selected color checker's patch count.
- Cleaned up unused functions and variables related to color checker handling.
- Ensured proper memory cleanup in the gui_cleanup function to prevent memory leaks.

Jiyone committed:
Code cleaning and added some comments

Jiyone committed:
Invert .cht and color definition lists
to add .cht files with built-ins and CGATS into their own combobox

Jiyone committed:
Changed the folder name for CGATS and .cht files.

Jiyone committed:
Code cleanup and improvements
- Optimization of the error message on the checker.
- Added a check for XYZ and Lab columns in the CGATS file.
- Optimization of some error messages.

Jiyone committed:
Initialization of values in checker structure

Jiyone committed:
Improved tooltip messages

Jiyone committed:
CGATS: improve color checker detection and messages
